sql/core/src/main/scala/org/apache/spark/sql/execution/GroupedIterator.scala

```diff
@@ -83,11 +83,12 @@ class GroupedIterator private(
 
   /** Holds a copy of an input row that is in the current group. */
   var currentGroup = currentRow.copy()
-  var currentIterator: Iterator[InternalRow] = null
+
   assert(keyOrdering.compare(currentGroup, currentRow) == 0)
```

Comment:
This is the whole row, not just the key.  This allows us to do the equality check on the key columns only (which might short circuit) instead of doing a full projection on each row to extract the key columns.